src/EditorFeatures/Core/SettingsData/Analyzers/AnalyzerSettingsUpdaterService.cs

```diff
@@ -0,0 +1,433 @@
+        {
+            return AddMissingRule(
+                result,
+                lastValidHeaderSpanEnd,
+                lastValidSpecificHeaderSpanEnd,
+                _language: LanguageNames.CSharp,
+                _diagnostic: diagnostic,
+                _newSeverity: severity.ToEditorConfigString(),
+                _isPerLanguage: true);
+        }
+
+        private static SourceText? AddMissingRule(
+            SourceText result,
+            TextLine? lastValidHeaderSpanEnd,
+            TextLine? lastValidSpecificHeaderSpanEnd,
+            string _language,
+            DiagnosticDescriptor _diagnostic,
+            string _newSeverity,
+            bool _isPerLanguage,
+            string? _categoryToBulkConfigure = null,
+            string? _optionNameOpt = null,
+            string? _newOptionValueOpt = null)
+        {
+            // Create a new rule configuration entry for the given diagnostic ID or bulk configuration category.
+            // If optionNameOpt and optionValueOpt are non-null, it indicates an option based diagnostic ID
+            // which can be configured by a new entry such as: "%option_name% = %option_value%:%severity%
+            // Otherwise, if diagnostic is non-null, it indicates a non-option diagnostic ID,
+            // which can be configured by a new entry such as: "dotnet_diagnostic.<%DiagnosticId%>.severity = %severity%
+            // Otherwise, it indicates a bulk configuration entry for default severity of a specific diagnostic category or all analyzer diagnostics,
+            // which can be configured by a new entry such as:
+            //  1. All analyzer diagnostics: "dotnet_analyzer_diagnostic.severity = %severity%
+            //  2. Category configuration: "dotnet_analyzer_diagnostic.category-<%DiagnosticCategory%>.severity = %severity%
+            var newEntry = !string.IsNullOrEmpty(_optionNameOpt) && !string.IsNullOrEmpty(_newOptionValueOpt)
+                ? $"{_optionNameOpt} = {_newOptionValueOpt}"
+                : _diagnostic != null
+                    ? $"{DiagnosticOptionPrefix}{_diagnostic.Id}{SeveritySuffix} = {_newSeverity}"
+                    : _categoryToBulkConfigure == AllAnalyzerDiagnosticsCategory
+                        ? $"{BulkConfigureAllAnalyzerDiagnosticsOptionKey} = {_newSeverity}"
+                        : $"{BulkConfigureAnalyzerDiagnosticsByCategoryOptionPrefix}{_categoryToBulkConfigure}{SeveritySuffix} = {_newSeverity}";
+
+            // Insert a new line and comment text above the new entry
+            var commentPrefix = _diagnostic != null
+                ? $"{_diagnostic.Id}: {_diagnostic.Title}"
+                : _categoryToBulkConfigure == AllAnalyzerDiagnosticsCategory
+                    ? "Default severity for all analyzer diagnostics"
+                    : $"Default severity for analyzer diagnostics with category '{_categoryToBulkConfigure}'";
+
+            newEntry = $"\r\n# {commentPrefix}\r\n{newEntry}\r\n";
```

Comment:
Consider determining end of line based on `end_of_line` setting (if exists) or based on the OS.